Tipster/MainWindow.cpp

```diff
@@ -87,6 +107,31 @@ for Haiku");
 			
 			break;
 		}
+		case UPDATE_ICON:
+		{
+			status_t status = msg->FindString("url", url);
+			
+			if (status == B_OK) {
+				BString artwork;
+				
+				msg->FindString("artwork", &artwork);
+				
+				BPath path("artwork");
+				path.Append(artwork);
+				
+				icon_bitmap = BTranslationUtils::GetBitmap(path.Path());
```

Comment:
As Scott commented already on the GCI task, you should use ressources from an rdef instead.